### ISSUE TITLE
Error in the MPPT algorithm

### ISSUE THREAD

**Issue by LucasAzulay** [spacelab-ufsc/eps2#184]
![WhatsApp Image 2023-03-23 at 09 48 34](https://user-images.githubusercontent.com/119054195/227269588-1ae34e62-bdbc-424d-8208-3a9e8d0dddda.jpeg)

**Comment by joaoclaudioeb** [comment]
@LucasAzulay, could you verify if CONFIG_SET_DUMMY_EPS is set to "0"? This macro is defined in the config.h file. Please, see the config.h that you are using during tests and let me know.

**Comment by joaoclaudioeb** [comment]
@LucasAzulay, we were able to reproduce the error by doing two things.

1) If the line 59 in config.h is presented as:

#define CONFIG_SET_DUMMY_EPS    1

It will generate the error.

2) If in config.h the macro CONFIG_SET_DUMMY_EPS was set to "0", but in the eps2_data.c the line 226 was modified to:  

#if CONFIG_SET_DUMMY_EPS == 0

It will generate the error, too.

So, please, could you try to use the code from the master's branch? Make a "git pull" to make sure to use the latest release of master (v0.4).

**Comment by LucasAzulay** [comment]
We are having some difficulty testing battery charging in EPS. We used a setup with a bench power supply connected (as shown in the diagram) to generate voltage in boost channel 1. We noticed that when connecting the source with 4.8 V, the current drops to zero. We turned on a small portable oscilloscope (we noticed that our bench oscilloscope has network noise) and we noticed the PWM with a dutty cycle starting at approximately 50% and after a while falling to less than 10%, reaching zero. In the videos it is possible to observe.

https://github.com/spacelab-ufsc/eps2/assets/119054195/881ac123-bb41-4e47-8453-9eee058744b9
https://github.com/spacelab-ufsc/eps2/assets/119054195/bbf43076-853a-4b2f-8572-97134da2b9a4
![image](https://github.com/spacelab-ufsc/eps2/assets/119054195/81bcedec-5552-4e7a-93cd-da5270821133)

The batteries used in this video are 3.9 V each. We tested with two batteries with 3.4 V each and observed the same thing.

If the batteries are less than 3.4V, we notice that the EPS neither starts nor generates the boost PWM.

**Comment by LucasAzulay** [comment]
We are having some difficulty testing battery charging in EPS. We used a setup with a bench power supply connected (as shown in the diagram) to generate voltage in boost channel 1. We noticed that when connecting the source with 4.8 V, the current drops to zero. We turned on a small portable oscilloscope (we noticed that our bench oscilloscope has network noise) and we noticed the PWM with a dutty cycle starting at approximately 50% and after a while falling to less than 10%, reaching zero. In the videos it is possible to observe.

https://github.com/spacelab-ufsc/eps2/assets/119054195/b88058d4-e97a-4c65-bd20-5d8c150b7e65
https://github.com/spacelab-ufsc/eps2/assets/119054195/cf8b3771-3430-4a40-a38b-be3a0ebc4a85
![image](https://github.com/spacelab-ufsc/eps2/assets/119054195/086a3b4d-20db-44e6-a691-9995f6e1f58b)


The batteries used in this video are 3.9 V each. We tested with two batteries with 3.4 V each and observed the same thing.

If the batteries are less than 3.4V, we notice that the EPS neither starts nor generates the boost PWM.
